libs/domain/picking/src/components/picking/picking.component.ts

```diff
@@ -0,0 +1,114 @@
+import { resolve } from '@spryker-oryx/di';
+import { asyncState, i18n, observe, valueType } from '@spryker-oryx/utilities';
+import { html, LitElement, TemplateResult } from 'lit';
+import { property } from 'lit/decorators.js';
+import { repeat } from 'lit/directives/repeat.js';
+import { BehaviorSubject, distinctUntilChanged, map, switchMap } from 'rxjs';
+import { ItemsFilters } from '../../models';
+import { PickingListService } from '../../services';
+
+export class PickingComponent extends LitElement {
+  @property({ attribute: 'picking-id' })
+  pickingId?: string;
+
+  @observe()
+  protected pickingId$ = new BehaviorSubject(this.pickingId);
+
+  protected pickingListService = resolve(PickingListService);
+
+  protected pickingList$ = this.pickingId$.pipe(
+    distinctUntilChanged(),
+    switchMap((id) => this.pickingListService.getById(id ?? ''))
+  );
+
+  //TODO - filter by item.status when it's there again
+  @asyncState()
+  protected notPickedItems = valueType(
+    this.pickingList$.pipe(
+      map((list) =>
+        list?.items.filter((item) => item.status === ItemsFilters.NotPicked)
+      )
+    )
+  );
+
+  @asyncState()
+  protected pickedItems = valueType(
+    this.pickingList$.pipe(
+      map((list) =>
+        list?.items.filter(
+          (item) => item.numberOfPicked && item.status === ItemsFilters.Picked
+        )
+      )
+    )
+  );
+
+  @asyncState()
+  protected notFoundItems = valueType(
+    this.pickingList$.pipe(
+      map((list) =>
+        list?.items.filter(
+          (item) =>
+            item.numberOfNotPicked ||
+            (item.numberOfPicked < item.quantity &&
+              item.status === ItemsFilters.NotFound)
+        )
+      )
+    )
+  );
+
+  protected tabs = [
+    {
+      id: ItemsFilters.NotPicked,
+      title: 'not-picked',
+      items: this.notPickedItems,
+    },
+    { id: ItemsFilters.Picked, title: 'picked', items: this.pickedItems },
+    {
+      id: ItemsFilters.NotFound,
+      title: 'not-found',
+      items: this.notFoundItems,
+    },
+  ];
+
+  protected override render(): TemplateResult {
+    this.tabs[0].items = this.notPickedItems;
```

Comment:
Not a good decision to update something inside render method. It will perform this on each re-render, even if nothing changes in the data. Due to each tab contains just filtered items, you can perform that filtering just in tabs configuration:

```
protected pickingList$ = this.pickingId$.pipe(
    distinctUntilChanged(),
    switchMap((id) => this.pickingListService.getById(id ?? ''))
);

protected tabs$ = this.pickingList$.pipe(map(list => ([
    {
      id: ItemsFilters.NotPicked,
      title: 'not-picked',
      items: list?.items.filter((item) => item.status === ItemsFilters.NotPicked),
    },
    { id: ItemsFilters.Picked, 
       title: 'picked', 
       items: list?.items.filter(item => item.numberOfPicked && item.status === ItemsFilters.Picked),
    },
    {
      id: ItemsFilters.NotFound,
      title: 'not-found',
      filter: list?.items.filter(item => item.numberOfNotPicked || (item.numberOfPicked < item.quantity && item.status === ItemsFilters.NotFound)),
    },
  ])));

@asyncState()
protected tabs = valueType(this.tabs$);
```

Reply:
Shame on me, I didn't test my suggestion. If something is unclear or broker - let me know